SampleApps/ReceiverDemoSwift/DPLReceiverSwiftAppDelegate.swift

```diff
@@ -6,29 +6,45 @@ class DPLReceiverSwiftAppDelegate: UIResponder, UIApplicationDelegate {
     var window: UIWindow?
     lazy var router: DPLDeepLinkRouter = DPLDeepLinkRouter()
 
-    func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool {
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
         
-        // Register a class to a route using object subscripting
-        self.router["/product/:sku"] = DPLProductRouteHandler.self
+        // Register a block to a route (matches dpl:///product/93598)
+        self.router.registerRoute("/product/:sku") { link in
+            
+            if let rootViewController = application.keyWindow?.rootViewController as? UINavigationController {
```

Comment:
There should use `guard let` and you might want to use optional chaining so you don’t need multiple `let`s